src/model/entities/rundown.ts

```diff
@@ -543,6 +546,151 @@ export class Rundown extends BasicRundown {
     this.markNextPart()
   }
 
+  public setNextFromDirection(direction: SetNextDirection, owner?: Owner): void {
+    this.assertActive(this.setNextFromDirection.name)
+
+    switch (direction) {
+      case SetNextDirection.PART_AFTER_NEXT_PART: {
+        this.setPartAfterCurrentNextPartAsNext(owner)
+        break
+      }
+      case SetNextDirection.PART_BEFORE_NEXT_PART: {
+        this.setPartBeforeCurrentNextPartAsNext(owner)
+        break
+      }
+      case SetNextDirection.SEGMENT_BEFORE_NEXT_SEGMENT: {
+        this.setFirstPartInSegmentBeforeNextSegmentAsNext(owner)
+        break
+      }
+      case SetNextDirection.SEGMENT_AFTER_NEXT_SEGMENT: {
+        this.setFirstPartInSegmentAfterNextSegmentAsNext(owner)
+        break
+      }
+    }
```

Comment:
```suggestion
    switch (direction) {
      case SetNextDirection.PART_AFTER_NEXT_PART: {
        return this.setPartAfterCurrentNextPartAsNext(owner)
      }
      case SetNextDirection.PART_BEFORE_NEXT_PART: {
        return this.setPartBeforeCurrentNextPartAsNext(owner)
      }
      case SetNextDirection.SEGMENT_BEFORE_NEXT_SEGMENT: {
        return this.setFirstPartInSegmentBeforeNextSegmentAsNext(owner)
      }
      case SetNextDirection.SEGMENT_AFTER_NEXT_SEGMENT: {
        return this.setFirstPartInSegmentAfterNextSegmentAsNext(owner)
      }
    }
```

Don't ask me why I prefer switch cases with returns, but I just do :D Just as a consideration for you of course :)

Reply:
Having `return` on same line as the method calls, might give the impression that the methods returns some sort of result - even though these don't :)

Reply:
They look nicer ;)

`return` in front of a method indicates that something is returned. Since all these methods return void I find it misleading to put a return statement in front of them.